### PR TITLE
[RDY] options: set 'history' to 10000 (the max) by default.

### PR DESCRIPTION
**Pull request by fmoralesc**
Re: https://github.com/neovim/neovim/issues/2676

### CODE REVIEW

**Comment by Pyrohh** [comment]
This LGTM, but...

@justinmk IIRC you said something along the lines of "it would be better if memory for history was allocated as needed, so we wouldn't need an arbitrary maximum". What do you think?

**Comment by fdinoff** [comment]
@Pyrohh if you had an arbitrary maximum you would probably need a setting that limits how much gets persisted between sessions. Persisting an ever growing history sounds like a bad idea.

**Comment by Pyrohh** [comment]
> Persisting an ever growing history sounds like a bad idea

Heh, yeah; this is why I think this PR is fine for now.

**Comment by fmoralesc** [comment]
I think work in the ShaDa feature will supersede this, so I hadn't pushed this further, but I'll merge if you think it's OK for now.

**Comment by Pyrohh** [comment]
> I think work in the ShaDa feature will supersede this, so I hadn't pushed this further

Maybe, I've no idea.

> ut I'll merge if you think it's OK for now.

It's OK with me.

**Comment by Pyrohh** [comment]
Like the other PRs changing defaults, a 2nd/3rd opinion would be helpful: @mhinz @justinmk what do you think?

File: runtime/doc/options.txt [hunk 1]
```diff
@@ -3552,12 +3552,12 @@ A jump table for the options with a short description can be found at |Q_op|.
...
 
 						*'history'* *'hi'*
-'history' 'hi'		number	(Vim default: 50, Vi default: 0)
+'history' 'hi'		number	(Vim default: 10000, Vi default: 0)
 			global
 	A history of ":" commands, and a history of previous search patterns
 	is remembered.  This option decides how many entries may be stored in
 	each of these histories (see |cmdline-editing|).
-	The maximum value is 10000.
+	Vim's default is the maximum value.
```
Comment (justinmk) [inline runtime/doc/options.txt]: This sentence seems confusing and could be removed.

Comment (mhinz) [inline runtime/doc/options.txt]: :+1:

**Comment by mhinz** [comment]
@Pyrohh  If there are no performance issues (you seemed to suggest that above), this should be merged.

**Comment by justinmk** [comment]
Merged. (I had this locally rebased last night but then lost connectivity)

**Comment by fmoralesc** [comment]
Thanks, @justinmk

**Pull request closed without merging**